otpInput/otpInput.js

```diff
@@ -0,0 +1,202 @@
+      input.value = '';
+      OtpInput.updateInput(instance);
+      return;
+    }
+
+    // create an array of characters
+    const chars = currentInput.value.split('');
+
+    let scheduledFocus = Math.min(instance.inputs.length - 1, i + chars.length);
+
+    // loop through the array and assign each character to the input
+    for (let pos = 0; pos < chars.length; pos++) {
+      if (pos >= instance.inputs.length) break;
+      let targetInput = instance.inputs[pos];
+
+      // paste value
+      targetInput.value = chars[pos];
+    }
+
+    // focus on the next input after pasting value
+    instance.inputs[scheduledFocus].focus();
+  }
+
+  static handleKeydown(instance, currentInput, i, e) {
+    const inputLength = instance.inputs.length;
+
+    if (e.keyCode === keycodes.backspace && i > 0) {
+      OtpInput.shiftInputValues(instance, i);
+      instance.inputs[i - 1].focus();
+      return;
+    }
+
+    if (e.keyCode === keycodes.delete && i !== inputLength - 1) {
+      OtpInput.shiftInputValues(instance, i + 1);
+      currentInput.select();
+      e.preventDefault();
+      return;
+    }
+
+    if (e.keyCode === keycodes.left) {
+      if (i > 0) {
+        e.preventDefault();
+        instance.inputs[i - 1].focus();
+        instance.inputs[i - 1].select();
+      }
+      return;
+    }
+
+    if (e.keyCode === keycodes.right) {
+      if (i + 1 < inputLength) {
+        e.preventDefault();
+        instance.inputs[i + 1].focus();
+        instance.inputs[i + 1].select();
+      }
+      return;
+    }
+  }
+
+  static shiftInputValues(instance, startIndex) {
+    for (let pos = startIndex; pos < instance.inputs.length - 1; pos++) {
+      instance.inputs[pos].value = instance.inputs[pos + 1].value;
+    }
+    instance.inputs[instance.inputs.length - 1].value = '';
+    OtpInput.updateInput(instance);
+  }
+
+  static updateInput(instance) {
+    let inputValue = Array.from(instance.inputs).reduce((acc, input) => {
+      acc += input.value.length ? input.value : ' ';
+      return acc;
+    }, '');
+    instance.hiddenInput.value = inputValue;
+    instance.readyToCheck = instance.inputs.every(
+      (input) => input.value !== ''
+    );
+
+    // Disable the submit button if readyToCheck is false
+    instance.submitButton.disabled = !instance.readyToCheck;
+  }
+
+  static checkForMatch(instance) {
+    if (instance.otp === instance.hiddenInput.value) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  submitOtp() {
+    let result;
+    if (OtpInput.getReadyToCheck(this)) {
+      if (OtpInput.checkForMatch(this)) {
+        this.resetForm();
+        result = true;
+      } else {
+        alert('Incorrect One Time Password (OTP)');
+        this.resetForm();
+        result = false;
+      }
+    } else {
+      alert('Please fill in all fields');
+    }
+    return result;
+  }
+
+  renderAndUse() {
+    document.addEventListener('DOMContentLoaded', () => {
+      OtpInput.createElements(this);
```

Comment:
Can't you just do `this.createElements()`?

Reply:
Yeah, that would likely work as well. If that's a preferred means I can change it. This came about from trying to think of a slightly more meaningful method name (although createElements isn't bad). It's either way for me.